### PR TITLE
[NIFI-12903] style updates for drag/drop UX

### PR DESCRIPTION
**Pull request by scottyaslan**
This PR creates reuseable styles for all cdk drag/drop UX. Please test with connection prioritizers as well as parameter context inheritence.

Kapture 2024-05-02 at 12 55 52

### CODE REVIEW

**Comment by mcgilman** [comment]
Will review...

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/connection/prioritizers/prioritizers.component.html [hunk 1]
```diff
@@ -16,53 +16,64 @@
...
+                    class="fa fa-info-circle"
+                    nifiTooltip
+                    [tooltipComponentType]="TextTip"
+                    tooltipInputData="Prioritizers that are available to reprioritize FlowFiles in this processors work queue."></i>
+            </div>
+            <div
+                class="cdk-list border"
+                cdkDropList
+                cdkDropListSortingDisabled
+                [cdkDropListDisabled]="isDisabled"
```
Comment (mcgilman) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/connection/prioritizers/prioritizers.component.html:33]: With this change, this `cdkDropList` is only shown when `!isDisabled` so we do not need to set `cdkDropListDisabled`.

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/connection/prioritizers/prioritizers.component.html [hunk 2]
```diff
@@ -73,20 +84,34 @@
...
+                @if (hasDescription(item)) {
+                    <i
+                        class="pl-1 fa fa-info-circle surface-color"
+                        nifiTooltip
+                        [tooltipComponentType]="TextTip"
+                        [tooltipInputData]="item.description"></i>
+                }
+            </div>
+            @if (canClose) {
+                <button class="pr-1" type="button" (click)="removeSelected(item, i)" [disabled]="isDisabled">
```
Comment (mcgilman) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/connection/prioritizers/prioritizers.component.html:100]: We no longer need to set `disabled` here since we only render the remove button when `!isDisabled`.

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/connection/prioritizers/prioritizers.component.html [hunk 3]
```diff
@@ -16,53 +16,64 @@
...
         </div>
-    </div>
+    }
     <div class="flex flex-col">
         <div>
-            Selected Prioritizers
+            @if (!isDisabled) {
+                Selected Prioritizers
+            } @else {
+                Prioritizers
```
Comment (mcgilman) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/connection/prioritizers/prioritizers.component.html:56]: When there are no selected prioritizers this is what the user sees. 

Screenshot 2024-05-03 at 9 31 36 AM

Since we are no longer rendering the source list, when there is nothing selected this looks similar to an `input`. Thoughts on handling this corner case explicitly so avoid this potential confusion?

**Review by mcgilman — COMMENTED** [review]
Thanks for the PR @scottyaslan! Looks great! Noted a few things below. I only commented them for the prioritizers, but my comments also apply to the parameter contexts since they are implemented similarly. I did not duplicate my comments.

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/connection/prioritizers/prioritizers.component.html (hunk 3 above)
Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/connection/prioritizers/prioritizers.component.html:55]: OK updated this UX to display "No value set".

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/parameter-contexts/ui/parameter-context-listing/parameter-context-inheritance/parameter-context-inheritance.component.ts [hunk 4]
```diff
@@ -73,7 +73,7 @@ export class ParameterContextInheritance implements ControlValueAccessor {
 
     protected readonly TextTip = TextTip;
 
-    isDisabled = false;
+    isDisabled = true;
```
Comment (mcgilman) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/parameter-contexts/ui/parameter-context-listing/parameter-context-inheritance/parameter-context-inheritance.component.ts:76]: Was this change intentional?

Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/parameter-contexts/ui/parameter-context-listing/parameter-context-inheritance/parameter-context-inheritance.component.ts:76]: no

**Review by mcgilman — APPROVED** [review]
Thanks for the updates @scottyaslan!
